### PR TITLE
grpc compiler.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,8 +9,6 @@ futures = "0.1"
 protobuf = "1.2"
 clippy = { version = "*", optional = true }
 
-[workspace]
```
Comment (BusyJay) [inline Cargo.toml]: Why remove this?

File: compiler/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+authors = [
+    "Stepan Koltsov <stepan.koltsov@gmail.com>",
+    "Neil Shen <overvenus@gmail.com>"
+]
+
+[dependencies]
+protobuf = "1.2"
+
+[[bin]]
+name = "protoc-gen-rust-grpc"
```
Comment (BusyJay) [inline compiler/Cargo.toml]: I prefer naming it `grpc_rust_plugin`.

File: compiler/src/util.rs [hunk 3]
```diff
@@ -0,0 +1,70 @@
+/// Adjust method name to follow the rust's style.
```
Comment (BusyJay) [inline compiler/src/util.rs]: Missing license.

**Comment by siddontang** [comment]
Please add rust-grpc license if we use this project.

File: compiler/Cargo.toml (hunk 2 above)
Comment (siddontang) [inline compiler/Cargo.toml]: I find two naming ways, `protoc-gen-grpc-xxx` or `grpc_xxx_plugin`.

Comment (disksing) [inline compiler/Cargo.toml]: I'll go with `grpc_rust_plugin`. The usage is 
```
protoc --grpc_out=. --plugin=protoc-gen-grpc=grpc_rust_plugin  *.proto
```

File: compiler/src/bin/grpc_rust_plugin.rs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+// Copyright 2017 PingCAP, Inc.
```
Comment (andelf) [inline compiler/src/bin/grpc_rust_plugin.rs]: should comes before orig old LICENSE,

File: compiler/src/util.rs (hunk 3 above)
Comment (andelf) [inline compiler/src/util.rs]: rust's style => rust-guidelines

File: compiler/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+[package]
+name = "compiler"
+version = "0.1.0"
+description = "gRPC compiler for grpc-rs"
+license = "MIT/Apache-2.0"
+authors = [
+    "Stepan Koltsov <stepan.koltsov@gmail.com>",
+    "Neil Shen <overvenus@gmail.com>"
```
Comment (siddontang) [inline compiler/Cargo.toml]: should remove this line.

**Comment by siddontang** [comment]
PTAL @BusyJay @andelf

File: compiler/Cargo.toml (hunk 5 above)
Comment (disksing) [inline compiler/Cargo.toml]: Why?

**Comment by siddontang** [comment]
Can it work now? If it can work, I think can merge it now.

LGTM

**Comment by siddontang** [comment]
PTAL @BusyJay @andelf

File: compiler/src/codegen.rs [hunk 6]
```diff
@@ -0,0 +1,484 @@
...
+                    w.field_entry("name", &self.fq_name());
+                });
+    }
+
+    // Method signatures
+    fn unary(&self, method_name: &str) -> String {
+        format!("{}(&self, req: {}) -> {}<{}>",
+                method_name,
+                self.input(),
+                fq_grpc("Result"),
```
Comment (BusyJay) [inline compiler/src/codegen.rs]: Please remote the result syntax

**Comment by siddontang** [comment]
PTAL @BusyJay @andelf

**Comment by BusyJay** [comment]
LGTM

File: compiler/src/codegen.rs (hunk 6 above)
Comment (siddontang) [inline compiler/src/codegen.rs]: any update @disksing ?

Comment (BusyJay) [inline compiler/src/codegen.rs]: I comment at the wrong line, the pr is ok now.

**Pull request merged**